File: order/src/main/java/com/ebuozturk/order/dto/product/Product.java
package com.ebuozturk.order.dto.product;

import com.fasterxml.jackson.annotation.JsonProperty;

public record Product(@JsonProperty("id") String id,
                      @JsonProperty("name") String name,
                      @JsonProperty("description") String description,
                      @JsonProperty("unitPrice") Double unitPrice,
                      @JsonProperty("quantityPerUnit") Integer quantityPerUnit,
                      @JsonProperty("unitsInStock") Integer unitsInStock
                      ) {
}

File: basket/src/main/java/com/ebuozturk/basket/dto/product/Product.kt
package com.ebuozturk.basket.dto.product;

import org.hibernate.annotations.GenericGenerator
import javax.persistence.*;

data class Product @JvmOverloads constructor(
    val id:String,
    val name: String,
    val description: String,
    val unitPrice: Double,
    val quantityPerUnit: Int,
    val unitsInStock: Int,

){

}
